### PR TITLE
feat: http handler support temp table.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1.	Implement a mechanism to clear data on S3 in case of a crash. Both MySQL and HTTP sessions need to write an ID to metadata with a TTL (time-to-live) and refresh it at regular intervals. The MySQL handler uses a daemon task for each connection, starting when the first query is made. The HTTP handler does it when renewing token.
2.	Both handlers need to clear data when a session ends to free up memory.
3.	The HTTP handler needs to clone the `temp_tbl_mgr` into the global ClientSessionManager and inject it into the session object when the next query arrives. 
4.	Consequently, the HTTP handler needs to release it (and clean up data) when a session times out.  only keep clones that are not empty, so we will not have to scan a lot of items when checking expiration

note for HTTP handler， will return error if  want to create temp table but token is not used (we need logout to free resouces).  waiting for https://github.com/datafuselabs/bendsql/pull/479 and modify to other clients

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16375)

### CODE REVIEW

File: src/query/service/src/auth.rs [hunk 1]
```diff
@@ -94,7 +94,6 @@ impl AuthMgr {
                     let user_info = user_api.get_user(&tenant, identity.clone()).await?;
                     session.set_authed_user(user_info, claim.auth_role).await?;
                 }
-                session.set_client_session_id(claim.session_id.clone());
```
Comment (SkyFan2002) [inline src/query/service/src/auth.rs:97]: Why is this removed?

Comment (youngsofun) [reply src/query/service/src/auth.rs:97]: duplicated: also set outside of call to this function

File: src/query/storages/common/session/src/temp_table.rs [hunk 2]
```diff
@@ -120,6 +133,9 @@ impl TempTblMgr {
                 true
             }
         };
+        if self.id_to_table.len() == 1 {
+            self.empty_state_changed = true;
```
Comment (SkyFan2002) [inline src/query/storages/common/session/src/temp_table.rs:137]: Is this redundant?

Comment (youngsofun) [reply src/query/storages/common/session/src/temp_table.rs:137]: no,  set to true when changing from empty to non-empty or reverse,  set to false when `is_empty` is called.
so caller add/remove mgr to map only when it is rue, do not have to lookup in map each time.

**Comment by SkyFan2002** [comment]
Let's track progress of http handler and clients in https://github.com/datafuselabs/databend/issues/16310

File: tests/sqllogictests/suites/temp_table/temp_memory_tables.sql [hunk 3]
```diff
@@ -28,7 +28,7 @@ create temp table t2(a int,b int)  engine=MEMORY;
 statement error 1005
 create temp table t2(a INT auto_increment) engine=MEMORY;
 
-statement error 1105
+statement error 1006
```
Comment (SkyFan2002) [inline tests/sqllogictests/suites/temp_table/temp_memory_tables.sql:31]: why is the ErrorCode changed? (just curious)

Comment (youngsofun) [reply tests/sqllogictests/suites/temp_table/temp_memory_tables.sql:31]: the format is `statement error <regex>`.
the output of mysql handler also contains an error code of mysql proto.

File: src/query/service/src/servers/http/v1/session/client_session_manager.rs [hunk 4]
```diff
@@ -69,14 +102,40 @@ impl ClientSessionManager {
...
+                    .collect::<Vec<_>>()
+            };
+            {
+                let mut guard = self.session_state.lock();
+                for (id, _) in expired.iter() {
+                    guard.remove(id);
+                }
+            }
+            for (_id, mgr) in expired {
+                drop_all_temp_tables(mgr).await.ok();
```
[the remark this answers is not in the document]
Comment (youngsofun) [reply src/query/service/src/servers/http/v1/session/client_session_manager.rs:133]: kv in meta expires themselves

**Review by SkyFan2002 — APPROVED** [review]
